src/bucket/Bucket.cpp

```diff
@@ -995,6 +999,64 @@ mergeCasesWithEqualKeys(MergeCounters& mc, BucketInputIterator& oi,
     ++ni;
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+bool
+Bucket::scanForEviction(AbstractLedgerTxn& ltx, EvictionIterator& iter,
+                        uint64_t& bytesToScan, uint32_t& maxEntriesToEvict,
+                        uint32_t ledgerSeq, medida::Meter& entriesEvictedMeter,
+                        medida::Counter& bytesScannedForEvictionMeter)
+{
+    ZoneScoped;
+    if (isEmpty())
+    {
+        // EOF
+        return false;
+    }
+
+    if (maxEntriesToEvict == 0 || bytesToScan == 0)
+    {
+        // Reached end of scan region
+        return true;
+    }
+
+    auto& stream = getStream();
+    stream.seek(iter.bucketFileOffset);
+
+    BucketEntry be;
+    while (stream.readOne(be))
+    {
+        if (be.type() == INITENTRY || be.type() == LIVEENTRY)
+        {
+            auto const& le = be.liveEntry();
+            if (isSorobanDataEntry(le.data) && isTemporaryEntry(le.data) &&
+                !isLive(le, ledgerSeq))
+            {
+                if (ltx.maybeEvict(le))
```

Comment:
Instead of modifying `LedgerTxn` to handle eviction, can you just - 
1. Load the entry from ltx.
2. Verify that the entry in ltx is expired.
3. Call erase.

I'm trying to reason if `LedgerTxn` needs to track `EntryChangeType`. If you just want the meta for entries deleted, I think you should be able to created a nested `LedgerTxn`, call `BucketManager::scanForEviction`, and then get the changes from that, which should only contain the meta for the evicted entries.

Reply:
The issue I was having is that whenever I tried to do some sort of nested `LedgerTxn`, the meta changes appeared as if they were TX based changes, which is incorrect. I'm not super knowledgeable on the `LedgerTxn` system though, so if there's a better way to hack it with a nested `LedgerTxn` I have no objections.

Reply:
Ok I have it working for the most part with a nested `LedgerTxn`, but testing is kinda weird. Another thing I noticed is that you assert for `LEDGER_ENTRY_REMOVED`, but when we scan to delete expired entries, we also emit a `LEDGER_ENTRY_STATE` (because I call `erase` in my version). The scan also updates the eviction iterator entry, so those changes will have to be ignored as well.